Products/ZenUI3/browser/eventconsole/export.py

```diff
@@ -75,11 +78,19 @@ def csv(self, response, archive, **params):
         response.setHeader('Content-Disposition', 'attachment; filename=events.csv')
         from csv import writer
         writer = writer(response)
-
+        events = [dict((k, v) for k, v in evt.iteritems() if v) for _, evt in
+                  self._query(archive, **params)]
+        keys = (x.iterkeys() for x in events)
+        fields = {item for item in itertools.chain.from_iterable(keys)}
```

Comment:
I'm thinking that `fields` should ultimately be a list rather than a set, e.g.
```
fields = list({item for item in itertools.chain.from_iterable(keys)})
```
It's still created from a set to get the necessary union of the fields.

I say this because it's converted to list when the header is written and also when the number of fields exceeds the `CSV_MAX_COLUMNS` value.  Just do the conversion once.